site/src/pages/TemplateVersionEditorPage/TemplateVersionEditorPage.tsx

```diff
@@ -357,10 +357,33 @@ const publishVersion = async (options: {
 	return Promise.all(publishActions);
 };
 
-const findInitialFile = (fileTree: FileTree): string | undefined => {
+const defaultMainTerraformFile = "main.tf";
+
+// findEntrypointFile function locates the entrypoint file to open in the Editor.
+// It browses the filetree following these steps:
+// 1. If "main.tf" exists in root, return it.
+// 2. Traverse through sub-directories.
+// 3. If "main.tf" exists in a sub-directory, skip further browsing, and return the path.
+// 4. If "main.tf" was not found, return the last reviewed "".tf" file.
+export const findEntrypointFile = (fileTree: FileTree): string | undefined => {
 	let initialFile: string | undefined;
 
-	traverse(fileTree, (content, filename, path) => {
+	if (Object.keys(fileTree).find((key) => key === defaultMainTerraformFile)) {
+		return defaultMainTerraformFile;
+	}
+
+	let skip = false;
+	traverse(fileTree, (_, filename, path) => {
+		if (skip) {
+			return;
+		}
+
+		if (filename === defaultMainTerraformFile) {
+			initialFile = path;
+			skip = true;
+			return;
+		}
```

Comment:
What is the order in which `traverse()` will traverse the nodes of `fileTree`? Are child paths sorted prior to being added to the frontier? In other words, if you have both `a/main.tf` and `z/main.tf`, which will get chosen? I know you have a test above, but is the output dependant on the ordering of the nodes? Is it just "first found wins"?

Reply:
Let me expand `traverse()` with sorting capabilities.

Reply:
Done

Reply:
nice 👍